src/render/utils/flat-tree.ts

```diff
@@ -0,0 +1,35 @@
+import { addUniqueItem, removeItem } from "../../utils/array"
+import { sortByDepth } from "./sort-by-depth"
+
+export interface DepthChild {
```

Comment:
This type does not have to be exported, and I think can be named better. In the FramerStudio codebase we have a pattern to prefix it with `With...` so something like `WithDepth` could work. Or `ChildWithDepth` if you want to be explicit.